src/services/maci/service.ts

```diff
@@ -684,6 +690,111 @@ export class MaciService {
     };
   }
 
+  async genProof({
+    signature,
+    message,
+    maciPrivKey,
+  }: IGenProofArgs): Promise<string> {
+    // get data from local storage
+    const maciData = await this.getMaciData();
+
+    const { id, isMerged } = await getPoll({
+      maciAddress: maciData.address,
+      signer: this.deployer,
+      provider: this.deployer.provider,
+    });
+
+    const pollId = Number(id);
+
+    const network = await this.getNetwork();
+    const tallyContractAddress = this.storage.mustGetAddress(
+      EContracts.Tally,
+      network,
+      `poll-${pollId}`,
+    );
+    console.log("isMerged:", isMerged, "id:", pollId);
+
+    if (!isMerged) {
+      // merge signups
+      try {
+        await mergeSignups({
+          pollId,
+          maciAddress: maciData.address,
+          signer: this.deployer,
+        });
+      } catch (error) {
+        console.log("Signups are already merged.", error);
```

Comment:
What if the error is something else? do we wanna start adding toasts for errors btw?

Reply:
I think we should just log the error now.